packages/http-server/src/http-server.ts

```diff
@@ -0,0 +1,114 @@
+// Copyright IBM Corp. 2017,2018. All Rights Reserved.
+// Node module: @loopback/http-server
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {createServer, Server, ServerRequest, ServerResponse} from 'http';
+import {AddressInfo} from 'net';
+import * as pEvent from 'p-event';
+
+export type HttpRequestListener = (
+  req: ServerRequest,
+  res: ServerResponse,
+) => void;
+
+/**
+ * Object for specifyig the HTTP / HTTPS server options
+ */
+export type HttpServerOptions = {
+  port?: number;
+  host?: string;
+  protocol?: HttpProtocol;
+};
+
+export type HttpProtocol = 'http' | 'https'; // Will be extended to `http2` in the future
+
+/**
+ * HTTP / HTTPS server used by LoopBack's RestServer
+ *
+ * @export
+ * @class HttpServer
+ */
+export class HttpServer {
+  private _port: number;
+  private _host?: string;
+  /**
+   * Protocol, default to `http`
+   */
+  private _protocol: HttpProtocol;
+  private _address: AddressInfo;
+  private httpRequestListener: HttpRequestListener;
+  private httpServer: Server;
+
+  /**
+   * @param httpServerOptions
+   * @param httpRequestListener
+   */
+  constructor(
+    httpRequestListener: HttpRequestListener,
+    httpServerOptions?: HttpServerOptions,
+  ) {
+    this._port = (httpServerOptions && httpServerOptions.port) || 0;
+    this._host = (httpServerOptions && httpServerOptions.host) || undefined;
+    this._protocol =
+      (httpServerOptions && httpServerOptions.protocol) || 'http';
+    this.httpRequestListener = httpRequestListener;
+  }
+
+  /**
+   * Starts the HTTP / HTTPS server
+   */
+  public async start(port?: number, host?: string) {
```

Comment:
Let's keep `start` parameterless please - both host and port should be configured at the time when the server is created. Unless there is a good justification why we need these two parameters?